packages/eas-cli/src/credentials/ios/appstore/authenticate.ts

```diff
@@ -18,22 +21,31 @@ import {
 
 const APPLE_IN_HOUSE_TEAM_TYPE = 'in-house';
 
+export enum AuthenticationMode {
+  API_KEY,
+  USER,
+}
+
 export type Options = {
   appleId?: string;
   teamId?: string;
+  teamName?: string;
+  teamType?: AppleTeamType;
+  ascApiKey?: MinimalAscApiKey;
   /**
    * Can be used to restore the Apple auth state via apple-utils.
    */
   cookies?: Session.AuthState['cookies'];
+  mode?: AuthenticationMode;
 };
 
 export type Team = {
   id: string;
-  name: string;
+  name?: string;
```

Comment:
```suggestion
    /** Name of the development team, this is undefined when ASC API keys are used instead of cookies for authentication. */
  name?: string;
```